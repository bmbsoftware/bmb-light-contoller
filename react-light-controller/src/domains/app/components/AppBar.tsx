import React from 'react';
import { withRouter } from 'react-router';
import { withStyles } from 'material-ui/styles';
import BottomNavigation from 'material-ui/BottomNavigation';
import BottomNavigationButton from 'material-ui/BottomNavigation/BottomNavigationButton';
import HomeIcon from 'material-ui-icons/Home';
import LightbulbOutlineIcon from 'material-ui-icons/LightbulbOutline';
import GroupIcon from 'material-ui-icons/Group';
import AutomationIcon from 'material-ui-icons/Alarm';

interface Props {
	location?: any;
	history?: any;
	classes: any;
}

interface State {
	current: number;
}

enum Locations {
	home = 1,
	lights = 2,
	groups = 3,
	automation = 4
}

const styles = (theme: any) => ({
	root: {
		width: '100%'
	}
});

class ButtonAppBar extends React.Component<Props, State> {
	constructor(props: Props) {
		super(props);
		const currentLocation = this.getLocation(location);
		this.state = {
			current: currentLocation
		};
	}

	public componentWillReceiveProps(nextProps: Props) {
		const { location } = nextProps;

		const currentLocation = this.getLocation(location);
		this.setState({
			current: currentLocation
		});
	}

	private getLocation = (location: any) => {
		switch (location.pathname) {
			case '/lights':
				return Locations.lights;
			default:
				return Locations.home;
		}
	}

	private handleChange = (event: any, value: number) => {
		const { history } = this.props;
		switch (value) {
			case Locations.lights:
				history.push('/lights');
				break;
			default:
				history.push('/');
		}

		this.setState({
			current: value
		});
	}

	public render() {
		const { classes } = this.props;
		const { current } = this.state;
		return (
			<div className={classes.root}>
				<BottomNavigation showLabels value={current} onChange={this.handleChange}>
					<BottomNavigationButton value={Locations.home} label="Home" icon={<HomeIcon />} />
					<BottomNavigationButton value={Locations.lights} label="Lights" icon={<LightbulbOutlineIcon />} />
					<BottomNavigationButton value={Locations.groups} label="Groups" icon={<GroupIcon />} />
					<BottomNavigationButton value={Locations.automation} label="Automation" icon={<AutomationIcon />} />
				</BottomNavigation>
			</div>
		);
	}
}

export default withRouter(withStyles(styles)(ButtonAppBar) as any);