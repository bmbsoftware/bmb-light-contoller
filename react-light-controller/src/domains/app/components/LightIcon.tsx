import React from 'react';
import { withStyles } from 'material-ui/styles';
import LightBulbIcon from 'material-ui-icons/LightbulbOutline';

interface Props {
	isLit: boolean;
	classes: any;
}

const styles = (theme: any) => {
	return {
		on: {
			color: '#f5dd07'
		},
		off: {
			color: '#0f0f0f'
		}
	};
};

export const LightIcon = (props: Props) => {
	const { isLit, classes } = props;
	const className = isLit ? classes.on : classes.off;
	return (
		<LightBulbIcon className={className} />
	);
};

export default withStyles(styles)(LightIcon);
