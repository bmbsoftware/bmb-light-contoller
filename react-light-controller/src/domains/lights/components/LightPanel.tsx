import React from 'react';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';

import LightIcon from 'domains/app/components/LightIcon';
import { Light } from 'domains/lights/model';

export interface Props {
	light: Light;
	className?: any;
	onClick?: (light: Light) => void;
}

export const LightPanel = (props: Props) => {
	const handleOnClick = () => {
		if (props.onClick) {
			props.onClick(props.light);
		}
	};

	return (
		<Card className={props.className}>
			<CardHeader
				onClick={handleOnClick}
				avatar={<LightIcon isLit={props.light.state.on} />}
				title={props.light.name}
			/>
		</Card>
	);
};

export default LightPanel;
