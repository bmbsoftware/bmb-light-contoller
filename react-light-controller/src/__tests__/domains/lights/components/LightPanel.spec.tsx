import React from 'react';
import { shallow } from 'enzyme';
import { LightPanel, Props } from 'domains/lights/components/LightPanel';

const props: Props = {
	light: {
		id: 1,
		name: 'Test Light',
		state: {
			on: true
		},
		type: '',
		swupdate: {
			state: 'updated',
			lastInstall: 'recently'
		}
	},

};

it('renders without crashing', () => {
	const wrapper = shallow(< LightPanel {...props} />);
	expect(wrapper).toMatchSnapshot();
});
