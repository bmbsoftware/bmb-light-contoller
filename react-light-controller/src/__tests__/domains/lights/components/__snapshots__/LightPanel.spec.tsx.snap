// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    light={
      Object {
        "id": 1,
        "name": "Test Light",
        "state": Object {
          "on": true,
        },
        "swupdate": Object {
          "lastInstall": "recently",
          "state": "updated",
        },
        "type": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <withStyles(CardHeader)
        avatar={
          <withStyles(Component)
            classes={Object {}}
            isLit={true}
          />
        }
        onClick={[Function]}
        title="Test Light"
      />,
      "className": undefined,
      "raised": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "avatar": <withStyles(Component)
          classes={Object {}}
          isLit={true}
        />,
        "onClick": [Function],
        "title": "Test Light",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <withStyles(CardHeader)
          avatar={
            <withStyles(Component)
              classes={Object {}}
              isLit={true}
            />
          }
          onClick={[Function]}
          title="Test Light"
        />,
        "className": undefined,
        "raised": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "avatar": <withStyles(Component)
            classes={Object {}}
            isLit={true}
          />,
          "onClick": [Function],
          "title": "Test Light",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
